packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanBar.tsx

```diff
@@ -86,6 +88,23 @@ function SpanBar(props: TCommonProps) {
     setLabel(longLabel);
   };
 
+  let criticalPathWhenParentCollapsed = {
+    ViewStart: Infinity,
+    ViewEnd: -Infinity,
+  };
+  if (!isChildrenExpanded && criticalPath) {
+    criticalPathWhenParentCollapsed = criticalPath.reduce((accumulator, each) => {
+      const critcalPathViewBounds = getViewedBounds(each.section_start, each.section_end);
+      const criticalPathViewStart = critcalPathViewBounds.start;
+      const criticalPathViewEnd = critcalPathViewBounds.end;
+
+      accumulator.ViewStart = Math.min(accumulator.ViewStart, criticalPathViewStart);
+      accumulator.ViewEnd = Math.max(accumulator.ViewEnd, criticalPathViewEnd);
+
+      return accumulator;
+    }, criticalPathWhenParentCollapsed);
+  }
```

Comment:
I feel like this is overly complicated and mixes view & data concerns. This is how I would do it:
  * if children are collapsed, re-compute the `criticalPath` array by merging adjacent segments (and btw I am not seeing where you check that they are in fact adjacent, so you're making an assumption that it's a contiguous area, an assumption that is not necessary in this piece of code)
  * no changes / bifurcation would be required in the view code below, it still works off a single `criticalPath` array